core/src/main/java/com/linecorp/armeria/client/RefreshingAddressResolverGroup.java

```diff
@@ -115,10 +116,12 @@
         } else {
             dnsRecordTypes = dnsRecordTypes(resolvedAddressTypes);
         }
+        cache = buildCache(cacheSpec != null ? cacheSpec
+                                             : requireNonNull(Flags.dnsCacheSpec(), "cacheSpec"));
```

Comment:
nit: Probably better to take non-null `cacheSpec` here. I think `Flags.dnsCacheSpec()` could be used before creating this class.

Reply:
I've changed the `cacheSpec` as nonnull. 👍